File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.java
class Solution {
    public int[] canSeePersonsCount(int[] heights) {
        Stack<Integer> stack = new Stack<>();
        int[] ans = new int[heights.length];

        for(int i=heights.length-1; i>=0; i--){
            while(!stack.isEmpty() && stack.peek() < heights[i]){
                stack.pop();
                ans[i]++;
            }

            if(!stack.isEmpty()){
                ans[i]++;
            }

            stack.push(heights[i]);
        }

        return ans;
    }
}

File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.cpp
class Solution {
public:
    vector<int> canSeePersonsCount(vector<int>& heights) {
        int n = heights.size();
        vector<int> ans(n, 0);
        stack<int> st;

        for (int i = n - 1; i >= 0; i--) {
            while (!st.empty() && st.top() < heights[i]) {
                st.pop();
                ans[i]++;
            }

            if (!st.empty()) {
                ans[i]++;
            }

            st.push(heights[i]);
        }

        return ans;
    }
};